### PR TITLE
Change core implementation to use all cores

### LINKED ISSUE #33: Force-spawn on game start
fix for Centrifuge

### LINKED ISSUE #34: Alter core-zone check to check all cores rather than center
for centrifuge..also

### PR DESCRIPTION
**Pull request by nilederg**
Rather than one pre-selected core at center (Breaks on certain maps), we check either all cores, or one (guaranteed real) core.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -32,3 +32,4 @@ jar{
         include "plugin.json"
     }
 }
+targetCompatibility = JavaVersion.VERSION_1_9
```
Comment (bend-n) [inline build.gradle]:
```suggestion
```

File: src/main/PlagueMain.java [hunk 2]
```diff
@@ -212,12 +212,20 @@ public void init() {
...
             // Blocks placement if player is survivor and attempting to place a core
-            // creation block, and that block is too close to plague core
-            if (distanceToCore < world.height() / 3.6 && action.player.team() != Team.malis) {
-                action.player.sendMessage("[scarlet]Cannot place core/vault that close to plague!");
-                return false;
+            // creation block, and that block is too close to any plague core
+            for (int i = 0; i < Team.malis.cores().size; i ++) {
+                CoreBuild plagueCore = Team.malis.cores().get(i);
+                float distanceToCore = new Vec2(action.tile.x, action.tile.y).dst(plagueCore);
+                // f*** s*** up if you try to build a core within a quarter of the world size
```
Comment (bend-n) [inline src/main/PlagueMain.java]:
```suggestion
```

File: src/main/PlagueMain.java [hunk 3]
```diff
@@ -212,12 +212,20 @@ public void init() {
...
+                if (distanceToCore < coringDistance && action.player.team() != Team.malis) {
+                    action.player.sendMessage("[scarlet]Cannot place core/vault that close to plague!\n" +
+                            "The limit is " + coringDistance + " tiles");
```
Comment (bend-n) [inline src/main/PlagueMain.java]:
```suggestion
                    action.player.sendMessage("[scarlet]Cannot place core/vault that close to plague!");
```

**Review by bend-n — CHANGES_REQUESTED** [review]

File: src/main/PlagueMain.java [hunk 4]
```diff
@@ -212,12 +212,18 @@ public void init() {
...
+            // Geometric mean for area-based quality
+            int coringDistance = (int) (Math.sqrt(world.height() * world.width()) * 0.25);
             // Blocks placement if player is survivor and attempting to place a core
-            // creation block, and that block is too close to plague core
-            if (distanceToCore < world.height() / 3.6 && action.player.team() != Team.malis) {
-                action.player.sendMessage("[scarlet]Cannot place core/vault that close to plague!");
-                return false;
+            // creation block, and that block is too close to any plague core
+            for (int i = 0; i < Team.malis.cores().size; i ++) {
+                CoreBuild plagueCore = Team.malis.cores().get(i);
```
Comment (bend-n) [inline src/main/PlagueMain.java:221]:
```suggestion
            Team.malis.cores().each(plagueCore -> {
```

File: src/main/PlagueMain.java [hunk 5]
```diff
@@ -212,12 +212,18 @@ public void init() {
...
+                float distanceToCore = new Vec2(action.tile.x, action.tile.y).dst(plagueCore);
+                if (distanceToCore < coringDistance && action.player.team() != Team.malis) {
```
Comment (bend-n) [inline src/main/PlagueMain.java]:
```suggestion
                if (distanceToCore < coringDistance) {
```
already handled

File: src/main/PlagueMain.java [hunk 6]
```diff
@@ -212,12 +212,17 @@ public void init() {
...
-                action.player.sendMessage("[scarlet]Cannot place core/vault that close to plague!");
-                return false;
+            // creation block, and that block is too close to any plague core
+            Team.malis.cores().each(plagueCore -> {
+                float distanceToCore = new Vec2(action.tile.x, action.tile.y).dst(plagueCore);
+                if (distanceToCore < coringDistance) {
+                    action.player.sendMessage("[scarlet]Cannot place core/vault that close to plague!");
+                    return false;
+                }
             }
```
Comment (bend-n) [inline src/main/PlagueMain.java:227]:
```suggestion
            });
```

Comment (nilederg) [reply src/main/PlagueMain.java:227]: huh

File: src/main/PlagueMain.java [hunk 7]
```diff
@@ -212,12 +212,18 @@ public void init() {
...
+            int coringDistance = (int) (Math.sqrt(world.height() * world.width()) * 0.25);
             // Blocks placement if player is survivor and attempting to place a core
-            // creation block, and that block is too close to plague core
-            if (distanceToCore < world.height() / 3.6 && action.player.team() != Team.malis) {
-                action.player.sendMessage("[scarlet]Cannot place core/vault that close to plague!");
-                return false;
+            // creation block, and that block is too close to any plague core
+            for (int i = 0; i < Team.malis.cores().size; i ++) {
+                CoreBuild plagueCore = Team.malis.cores().get(i);
+                float distanceToCore = new Vec2(action.tile.x, action.tile.y).dst(plagueCore);
```
Comment (bend-n) [inline src/main/PlagueMain.java:222]:
```suggestion
                float distanceToCore = new Vec2(action.tile.x, action.tile.y).dst(new Vec2(plagueCore.tile.x, plagueCore.tile.y));
```

**Pull request closed without merging**